### ISSUE TITLE
NodeMCO/ESP Support

### ISSUE THREAD

**Issue by ozarchie** [mayaculpa/hapi#20]
I have created hapifirmware from the original firmware directories. Using one file, you can now compile for USB, ENET or ESP. The  firmware supports the NodeMCU (which is what I tested on) and the original 2560 (I have only tested the USB as I don’t have the ENET shield).
I could not get the pull request to work so I am attaching the files to this issue. 

Unfortunately, I cannot get the smart module to work on a RasPi, but I have documented the steps I tried in the attached odt file.
[hapifirmware.ino.zip](https://github.com/mayaculpa/hapi/files/960241/hapifirmware.ino.zip)

[Hapi-Raspi-install.odt.zip](https://github.com/mayaculpa/hapi/files/960243/Hapi-Raspi-install.odt.zip)

. I do not know the function of influxdb and sqlbrowser and sqlite3 do not recognize the sql files. Some files are missing? e.g. create_tables.sql.

I will check back in the future - maybe when there is more documentation - and continue testing on NodeMCU and maybe VROOM32

**Comment by TylerReedMC** [comment]
Can I make a couple of requests for the NodeMCU work?

**Comment by ozarchie** [comment]
Hi Tyler. No problems.
I have been using the NodeMCU for automation of my chicken coop. The
changes that I made to the dumb module to get NodeMCU support were in
preparation for integration of the vegetable garden attached to it.
 I have light Sensors, moisture sensors,  pH,  TDS, DHT and 18b20 devices.
I am happy to contribute.
My problem with github was with the desktop version. I will switch to the
online version.
Cheers Archie

**Comment by ozarchie** [comment]
Hi Tyler,
I installed the current code base on a Pi3.
It is obvious that it is under heavy development, and that the HapiListener from September last year is migrating to a local/cloud based MQQT implementation.
As such, the current arduino telnet/usb implementation will probably change to a MQQT based data publisher. I tried to extract the project communication architecture from smart_module.py but I am not that familiar with the intricacies of Python.
I am happy to help when that interface is defined. 
Cheers
Archie
